packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -140,12 +141,14 @@ export class Summarizer extends EventEmitter implements ISummarizer {
         onBehalfOf: string,
         options?: Readonly<Partial<ISummarizerOptions>>): Promise<SummarizerStopReason> {
         try {
+            assert(!this.running, "reentrancy");
+            this.running = true;
             return await this.runCore(onBehalfOf, options);
         } catch (error) {
             this.stop("summarizerException");
```

Comment:
>); [](http://example.com/codeflow?start=43&length=2)

do we need to forceClose == true here ?

Reply:
No, finally {} block will do it for both success & exception paths.